### ISSUE TITLE
Generic component wrapper to avoid content reflow in viewport

### ISSUE THREAD

**Issue by jamesbhobbs** [BBC-archive/psammead#2279]
**Is your feature request related to a problem? Please describe.**
We need a component that can load 3rd party content without breaking our coding conventions: https://github.com/bbc/simorgh/blob/latest/docs/Code-Standards.md the particularly troublesome one being content reflow avoidance.

**Describe the solution you'd like**
I suggest a generic highly reusable wrapper component which (in a way that meets as best as possible our browser support needs) does not every reflow content that is in a users viewport. To do this the logic is: 
if thing in viewport do not resize
if thing outside of viewport allow resize
    if below viewport do nothing else
    if above viewport adjust user scroll position so the content in the viewport is unchanged.

Broadly speaking this is not too tricky, presuming the css changes to the height of wrapper component and the scroll position are done at the same time. But on Opera mini in particular this is much more challenging.

NB we probably want to add to this component timeouts on the content loading - this does not need to be done for the first iteration but should probably be planned.

See suggestions of how in additional context below.

**Describe alternatives you've considered**
This would be used for Visual Journalism content, IDT (data) components and adverts. Given that there will be a very large number doing our best to avoid this problem is highly important and urgent as adverts will be blocking releases quite soon (technically they already are).

**Testing notes**
Third party components include ads,visual journalism or IDT content which are wrapped inside an iframe.This issue requires testing the components only on storybook for now.

Dev insight: So much x-browser and x-device testing. I suggest we ensure that content is accessible as a fallback to rendering correctly, i.e. it is better for a thing in the wrapper component to render in a permanently ugly way that to disrupt the reading experience by vertically reflowing.

- [x] This feature is expected to need manual testing.

**Additional context**
NB although this refers to vertical reflowing it should also be enabled for horizontal reflowing.

**_In general I suggest the following logic:_**
Display overflow by default on the wrapper with a scrollbar i.e. it's box size is not define by its children.

Check if thing is in viewport and if it's not whether it's above or below the viewport.

If not in viewport and below it, let it resize and remove the scroll bar.
If not in viewport and above it, resize it while at the same time adjusting the scroll position by the amount changed so there is no perciptible difference.
If it is in the viewport do nothing.

NB you need to accommodate scrolling up and down and entry point on the page that do not presume the user loads the page at the top of the document, e.g. they are linked to an anchor/id lower down the page.

More detail on implementation: the wrapper component should have an optional prop, it default to 0 and may be given any value in px. This value is the min-height of the component. This will be useful when we know for a given use of the wrapper that it will almost always be "N"px high - therefore reserving the space we want it to have - we should have same logic for width.

In case the maths on the change in size of the element is wrong it might be an idea to add a css transition. But if it's correct I'm not sure if this will slowly transition creating a perceived reflow.

**_Resources for how to do this:_**
https://stackoverflow.com/questions/2481350/how-to-get-scrollbar-position-with-javascript

Set min-height and add transition if height changes: https://css-tricks.com/content-jumping-avoid/

Show ad but hide its overflow: https://css-tricks.com/findingfixing-unintended-body-overflow/

Get current things in viewport: https://stackoverflow.com/questions/2481350/how-to-get-scrollbar-position-with-javascript

Scroll to position top https://caniuse.com/#feat=mdn-api_scrolltooptions_top

NB this example which implements a similar but much more simplistic solution to a subset of the problem described https://itnext.io/how-to-stop-content-jumping-when-images-load-7c915e47f576#913e It includes important details such as ensuring it's executed only once.

Check if element is in the viewport: https://gomakethings.com/how-to-test-if-an-element-is-in-the-viewport-with-vanilla-javascript/

Opera mini needs a bookmark to scroll to, so add one to top of element at top of viewport in order to then link, see second answer on https://stackoverflow.com/questions/21174555/trigger-scroll-in-opera-mini oh and you might need this to check what's in the viewport to opera mini https://gomakethings.com/how-to-test-if-an-element-is-in-the-viewport-with-vanilla-javascript/

UA sniffing for edge cases: https://dev.opera.com/articles/fixing-the-scrolltop-bug/

Nice sort of related doc: https://javascript.info/size-and-scroll
https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollHeight
